README.md
<b>STT Collection Manager</b> ![Logo](https://github.com/augustenz/STT-Collection-Manager/blob/master/src/sttcollectionmanager/resources/logo48x48.png)

STT Collection Manager is a desktop Java application, developed to help manage a crew collection for Disruptor Beam's Star Trek Timelines game.

<img src="https://github.com/augustenz/STT-Collection-Manager/blob/master/AllCrew.png" width="15%" height="15%" /><img src="https://github.com/augustenz/STT-Collection-Manager/blob/master/Collection.png" width="15%" height="15%" /><img src="https://github.com/augustenz/STT-Collection-Manager/blob/master/SearchCollection.png" width="15%" height="15%" /><img src="https://github.com/augustenz/STT-Collection-Manager/blob/master/Charts.png" width="15%" height="15%" /><img src="https://github.com/augustenz/STT-Collection-Manager/blob/master/ChartsWindow.png" width="15%" height="15%" />

<b>Download the latest version: </b><a href="https://github.com/augustenz/STT-Collection-Manager/raw/master/STTCollectionManager_v1.2.zip">STTCollectionManager_v1.2.zip</a>

[![Donate](https://github.com/augustenz/STT-Collection-Manager/blob/master/src/sttcollectionmanager/resources/beer32x32.png)](https://www.paypal.me/augustenz) [![Donate](https://img.shields.io/badge/Donate-PayPal-green.svg)](https://www.paypal.me/augustenz) <-- <a href="https://www.paypal.me/augustenz">If you like this Software feel free to buy me a beer! :)</a>

<hr>

<h2>Features</h2>

- View all crew that is in the game in table format
- Add crew to your collection
- Remove crew from your collection
- Update the crew that is in game whenever new crew has been added
- Sort by any column
- Filter both tables using the Search function
- Export the all crew table or the collection table to a HTML, Text or Spreadsheet file
- Import crew data for the all crew table or the collection table from a Spreadsheet file
- Update the stats (number of times fused, current level, skills, fully equipped, in vault) of the crew in the collection
- Press on any Character's Name, Race or Trait and open the corresponding Wiki page
- Change the font of the tables and the skin of the app
- View charts and stats about the crew.

<hr>

<h2>Changelog</h2>

<b>
16/10/2017
Version 1.2
</b><br>
<br>

New Features:
  - Added Settings tab (change every URL in the app)
  - Added Import function (import crew data from .xls files for both Collection and Crew Tables)
  - Added faster way to add copies to collection (press an already checked checkbox in the Crew table)
  - Clicking or tabbing table cells auto selects-all cell content for faster editing
  - Collection table can now be exported to HTML too
  - All table rows in exported HTML files are now colored by Crew rarity
  - HTML tables are now sortable by any column (clicking on table headers toggles Asc/Desc sort order)
  - Progress window with a text area and a progress bar for import tasks

Bugs/Performance
  - Fixed bug where the splash screen wouldn't show
  - Fixed bug where tabbing through the Collection cells would delete an entry
  - Workaround for a bug which would add the first member in the Collection twice (first run only)
  - Various other performance optimizations and bug fixes

<b>
11/8/2016
Version 1.1
</b><br>
<br>

- Added: Highlight row under cursor
- Stacked Bar colors changed
- 1-click cell edit
- Tab to next editable cell
- Crew in collection update fixes - improved performance

<b>
5/8/2016
Version 1.0 (Initial release)
</b><br>
<br>

- Removed table tooltips
- Fixed crew in collection SQL update (single quote escape)

<hr>

<h2>Help</h2>

STT Collection Manager v1.2<br>
Author: August Enzelberger<br>
Project Homepage: https://github.com/augustenz/STT-Collection-Manager<br>
Date: 16/10/2017<br>

STT Collection Manager is a Java application, developed to help manage a Crew collection for Disruptor Beam's Star Trek Timelines game.

<hr>

<b>Important Info</b>

This application needs the latest JRE (Java Runtime Environment) to run, it is available for various systems and can be obtained
from www.oracle.com (or https://java.com/en/download/).

JavaFX is used by this application to display charts, it is included in Oracle's JRE so nothing else is needed if that JRE is installed.
Other JREs (like OpenJDK in Linux) need extra packages (like OpenJFX in Linux) to run Java applications that contain JavaFX elements.
In that case either the extra packages can be installed, or Oracle's JRE.
JavaFX is required to run this application.

The database is saved in a folder with the name "STTCollectionManager" in the user's home directory, or in the directory containing the application.

To run the application, double-click on the STTCollectionManager.jar file, or right click and choose 
open with and your JRE (java.exe on Windows, OpenJDK or Oracle Java on Linux).

To run the project from the command line, go to the application folder and 
type the following:


java -jar "STTCollectionManager.jar"

The Shortcut icons folder contains icon images of various sizes if you wish to make a shortcut
for the application.

<hr>

STT Collection Manager consists of six tabs:

<b>1) The Collection tab</b>

This tab contains the Crew Collection table. Here all crew that is currently in the collection can be edited. The editable fields are all the skill fields,
the fused stars field, the level field, the quantity field (in case of multiples) and a checkbox that removes crew from the collection.
Clicking on an editable cell opens it for editing and auto selects-all content for faster editing.
While editing a cell, pressing Tab on the keyboard will move to the next cell.
Tabbing will cycle through all editable columns (except Quantity) and all rows.
Clicking on a column header will sort the table according to that column.
Clicking on the Crew Name field, Race field or any of the Traits in the traits field will open the corresponding wiki page.
Right-clicking on crew members brings up a menu that allows to make a copy of the selected crew member (multiples but with different stats).

<b>2) The Crew tab</b>

This tab contains the Crew table. Here all crew in the game is listed, no fields are editable here except a checkbox that allows adding crew to the
collection.
If this checkbox is already checked, additional clicks on the checkbox will create copies in the Collection (faster way for adding duplicates).
Clicking on a column header will sort the table according to that column.
Clicking on the Crew Name field, Race field or any of the Traits in the traits field will open the corresponding wiki page.

<b>3) The Log tab</b>

This tab contains a textarea where messages regarding the execution of commands can be seen.

<b>4) The Charts tab</b>

This tab contains four charts:

 - All Crew pie chart: A pie chart where the balance (quantity of crew/rarity) of the crew in the game can be seen. Each slice represents the number
of crew in game of a certain rarity.
 
 - Crew Collection pie chart: A pie chart where the balance (quantity of crew/rarity) of the collection can be seen. Each slice represents the number
of crew in the collection of a certain rarity.
 
 - Bar chart: A bar chart where on the left the bars represent the number of crew in the game of a certain rarity and on the right the number of crew
in the collection of a certain rarity.
 
 - Stacked bar chart: Similar to the bar chart, only that the bars are stacked instead of being side-by-side.

Every chart will show the number that each graphic element represents as a mouse tooltip.
If a chart is clicked it is opened in a new window where it is enlarged and can be maximized.

<b>5) The Stats tab</b>

This tab has a text field where the number of crew slots can be entered. It has some information about the collection, such as the number of crew/crew
slots, the number of crew in the collection per rarity/the number of crew in the game per rarity and the average crew level.

<b>6) The Settings tab</b>

This tab has three text fields where the URLs used in the app can be changed.

<hr>

<b>The Menu Bar</b>

The menu bar consists of seven menus: File, Search, Import, Export, Skin, Font and Help.

<b>1) File</b>

 - Download Crew: Drops the crew table (the collection is kept intact) and re fetches all the crew data from the wiki. This is useful when new crew is
added to the wiki.

 - Refresh Tables: Refreshes the data shown in the tables, affects only the view, not the data.

 - Exit: Saves various settings like the window size, window position etc, shuts down the database and closes the application.

<b>2) Search</b>

Using the search function, both tables can be filtered according to certain criteria. Typing in any text field will immediately query the database and
populate the tables accordingly.
Any info can be entered only partly and the matches (if any) will be shown.
Multiple traits can be entered (they can be separated by a space or a comma) but there will be matches only if they are entered in the same order that they
appear in.
Single traits of course will be matched regardless of their position in the field.

For the search window to open, this menu item has to be pressed when either the Collection tab or the Crew tab are open. It will filter the table that is
open.

<b>3) Import</b>

 - Import All Crew: Imports crew data from an .xls file (tab seperated text file) to the "All Crew" table.

 - Import Crew Collection: Imports crew data from an .xls file (tab seperated text file) to the "Crew Collection" table.

<b>4) Export</b>

 - Export All Crew: Exports all crew in the game to a HTML (crew in collection is shown in bold), text, or xls (tab seperated text file - good for Spreadsheets) file.

 - Export Crew Collection: Exports the crew collection to a HTML, text, or xls (tab seperated text file - good for Spreadsheets) file.

<b>5) Skin</b>

Lists skins available on the system and allows changing the current skin. This setting is saved.

6) Font

 - Increase Font Size: Increases the font size.
 - Decrease Font Size: Decreases the font size.
 - Reset Font Size: Resets the font size (12).
 - Change Font: Lists fonts found on the system and allows to change the font being used in the tables.

These settings are saved.

<b>7) Help</b>

 - Help: Probably this readme file :)
 - Info: Info about the system.
 - About: The application credits.

<hr>

<h2>Credits</h2>

Author: August Enzelberger<br>
Project Webpage: https://github.com/augustenz/STT-Collection-Manager<br>

STT Collection Manager was created with Netbeans IDE 8.1.

Crew data comes from the very awesome Star Trek Timelines Wiki (https://stt.wiki) and its awesome users.

App icon by Memory Alpha (http://memory-alpha.wikia.com).

Roddenberry font by Jaynz (http://www.fontspace.com/pixel-sagas/roddenberry).

Icons from the Noun Project (https://thenounproject.com): Info by Karthick Nagarajan, Export by Mourad Mokrane, Refresh by Alex Auda Samora, Download by Guillaume Bahri, Zoom out by useiconic.com, Zoom in by useiconic.com, Reset zoom by Leonardo Schneider, Font by iconsmind.com, Help by Dima Lagunov, Log by Arthur Shlain, Stats by Calvin Goodman, Charts by Prerak Patel, About by Gregor Črešnar, Skin by Rflor, Exit by Andres Gleixner, Tables by Viktor Vorobyev, Search by Iconfactory Team, Settings by Michal Beno.

JFontChooser by Masahiko Sawai (http://jfontchooser.osdn.jp/site/jfontchooser).

Nimbus checkbox alternate row background color bug fix by Harald Barsnes (https://github.com/compomics).

Many thanks to all the awesome people at Stack Overflow (http://stackoverflow.com) and Coderunch (https://coderanch.com), without all of the excellent information there this project would have been much harder to develop.

This software is released under the GNU General Public License, version 3 (https://www.gnu.org/licenses/gpl-3.0.html).

Game content and materials are trademarks and copyrights of their respective publisher and its licensors.

<hr>

<h2>License</h2>

STT Collection Manager
Copyright (C) 2016  August Enzelberger

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.

                    GNU GENERAL PUBLIC LICENSE
                       Version 3, 29 June 2007

 Copyright (C) 2007 Free Software Foundation, Inc. <http://fsf.org/>
 Everyone is permitted to copy and distribute verbatim copies
 of this license document, but changing it is not allowed.

                            Preamble

  The GNU General Public License is a free, copyleft license for
software and other kinds of works.

  The licenses for most software and other practical works are designed
to take away your freedom to share and change the works.  By contrast,
the GNU General Public License is intended to guarantee your freedom to
share and change all versions of a program--to make sure it remains free
software for all its users.  We, the Free Software Foundation, use the
GNU General Public License for most of our software; it applies also to
any other work released this way by its authors.  You can apply it to
your programs, too.

  When we speak of free software, we are referring to freedom, not
price.  Our General Public Licenses are designed to make sure that you
have the freedom to distribute copies of free software (and charge for
them if you wish), that you receive source code or can get it if you
want it, that you can change the software or use pieces of it in new
free programs, and that you know you can do these things.

  To protect your rights, we need to prevent others from denying you
these rights or asking you to surrender the rights.  Therefore, you have
certain responsibilities if you distribute copies of the software, or if
you modify it: responsibilities to respect the freedom of others.

  For example, if you distribute copies of such a program, whether
gratis or for a fee, you must pass on to the recipients the same
freedoms that you received.  You must make sure that they, too, receive
or can get the source code.  And you must show them these terms so they
know their rights.

  Developers that use the GNU GPL protect your rights with two steps:
(1) assert copyright on the software, and (2) offer you this License
giving you legal permission to copy, distribute and/or modify it.

  For the developers' and authors' protection, the GPL clearly explains
that there is no warranty for this free software.  For both users' and
authors' sake, the GPL requires that modified versions be marked as
changed, so that their problems will not be attributed erroneously to
authors of previous versions.

  Some devices are designed to deny users access to install or run
modified versions of the software inside them, although the manufacturer
can do so.  This is fundamentally incompatible with the aim of
protecting users' freedom to change the software.  The systematic
pattern of such abuse occurs in the area of products for individuals to
use, which is precisely where it is most unacceptable.  Therefore, we
have designed this version of the GPL to prohibit the practice for those
products.  If such problems arise substantially in other domains, we
stand ready to extend this provision to those domains in future versions
of the GPL, as needed to protect the freedom of users.

  Finally, every program is threatened constantly by software patents.
States should not allow patents to restrict development and use of
software on general-purpose computers, but in those that do, we wish to
avoid the special danger that patents applied to a free program could
make it effectively proprietary.  To prevent this, the GPL assures that
patents cannot be used to render the program non-free.

  The precise terms and conditions for copying, distribution and
modification follow.

                       TERMS AND CONDITIONS

  0. Definitions.

  "This License" refers to version 3 of the GNU General Public License.

  "Copyright" also means copyright-like laws that apply to other kinds of
works, such as semiconductor masks.

  "The Program" refers to any copyrightable work licensed under this
License.  Each licensee is addressed as "you".  "Licensees" and
"recipients" may be individuals or organizations.

  To "modify" a work means to copy from or adapt all or part of the work
in a fashion requiring copyright permission, other than the making of an
exact copy.  The resulting work is called a "modified version" of the
earlier work or a work "based on" the earlier work.

  A "covered work" means either the unmodified Program or a work based
on the Program.

  To "propagate" a work means to do anything with it that, without
permission, would make you directly or secondarily liable for
infringement under applicable copyright law, except executing it on a
computer or modifying a private copy.  Propagation includes copying,
distribution (with or without modification), making available to the
public, and in some countries other activities as well.

  To "convey" a work means any kind of propagation that enables other
parties to make or receive copies.  Mere interaction with a user through
a computer network, with no transfer of a copy, is not conveying.

  An interactive user interface displays "Appropriate Legal Notices"
to the extent that it includes a convenient and prominently visible
feature that (1) displays an appropriate copyright notice, and (2)
tells the user that there is no warranty for the work (except to the
extent that warranties are provided), that licensees may convey the
work under this License, and how to view a copy of this License.  If
the interface presents a list of user commands or options, such as a
menu, a prominent item in the list meets this criterion.

  1. Source Code.

  The "source code" for a work means the preferred form of the work
for making modifications to it.  "Object code" means any non-source
form of a work.

  A "Standard Interface" means an interface that either is an official
standard defined by a recognized standards body, or, in the case of
interfaces specified for a particular programming language, one that
is widely used among developers working in that language.

  The "System Libraries" of an executable work include anything, other
than the work as a whole, that (a) is included in the normal form of
packaging a Major Component, but which is not part of that Major
Component, and (b) serves only to enable use of the work with that
Major Component, or to implement a Standard Interface for which an
implementation is available to the public in source code form.  A
"Major Component", in this context, means a major essential component
(kernel, window system, and so on) of the specific operating system
(if any) on which the executable work runs, or a compiler used to
produce the work, or an object code interpreter used to run it.

  The "Corresponding Source" for a work in object code form means all
the source code needed to generate, install, and (for an executable
work) run the object code and to modify the work, including scripts to

CHANGELOG.txt
16/10/2017
Version 1.2

New Features:
  - Added Settings tab (change every URL in the app)
  - Added Import function (import crew data from .xls files for both Collection and Crew Tables)
  - Added faster way to add copies to collection (press an already checked checkbox in the Crew table)
  - Clicking or tabbing table cells auto selects-all cell content for faster editing
  - Collection table can now be exported to HTML too
  - All table rows in exported HTML files are now colored by Crew rarity
  - HTML tables are now sortable by any column (clicking on table headers toggles Asc/Desc sort order)
  - Progress window with a text area and a progress bar for import tasks

Bugs/Performance
  - Fixed bug where the splash screen wouldn't show
  - Fixed bug where tabbing through the Collection cells would delete an entry
  - Workaround for a bug which would add the first member in the Collection twice (first run only)
  - Various other performance optimizations and bug fixes

11/8/2016
Version 1.1

- Added: Highlight row under cursor
- Stacked Bar colors changed
- 1-click cell edit
- Tab to next editable cell
- Crew in collection update fixes - improved performance

5/8/2016
Version 1.0 (Initial release)

- Removed table tooltips
- Fixed crew in collection SQL update (single quote escape)

src/sttcollectionmanager/AboutJFrame.java
/*
 * Copyright (C) 2016 August Enzelberger <augustenz+github@gmail.com> (https://github.com/augustenz)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package sttcollectionmanager;

import java.awt.Desktop;
import java.awt.Image;
import java.awt.List;
import java.awt.image.BufferedImage;
import java.io.IOException;
import java.net.URI;
import java.net.URISyntaxException;
import java.util.ArrayList;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.imageio.ImageIO;
import javax.swing.ImageIcon;
import javax.swing.JOptionPane;
import javax.swing.event.HyperlinkEvent;
import javax.swing.event.HyperlinkListener;

/**
 *
 * August Enzelberger <augustenz+github@gmail.com> (https://github.com/augustenz)
 */
public class AboutJFrame extends javax.swing.JFrame {

    /**
     * Creates new form aboutJFrame
     */
    public AboutJFrame() {
        initComponents();
        Image image=null;
        ArrayList<BufferedImage> images = new ArrayList<BufferedImage>();
        try{
            List icons = new List();
            BufferedImage bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo16x16.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo18x18.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo20x20.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo24x24.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo32x32.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo36x36.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo40x40.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo48x48.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo64x64.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo96x96.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo128x128.png"));
            images.add(bi);
            bi = ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo256x256.png"));
            images.add(bi);
            ImageIcon imi = new ImageIcon(bi);
            icons.add(imi.toString());
            this.setIconImages(images);
        }catch( Exception e ) { 
            try {
                this.setIconImage(ImageIO.read(getClass().getResource("/sttcollectionmanager/resources/logo16x16.png")));
            } catch (IOException ex) {
                Logger.getLogger(STTCollectionManager.class.getName()).log(Level.SEVERE, null, ex);
            }
        }
        String text="<p><b>Author:</b> August Enzelberger (<a href=\"mailto:augustenz+github@gmail.com\">augustenz+github@gmail.com</a>)</p>"
                + "<p><b>Project Homepage:</b> <a href=https://github.com/augustenz/STT-Collection-Manager>https://github.com/augustenz/STT-Collection-Manager</a></p>"
                + "\n"
                + "<p>STT Collection Manager was created with Netbeans IDE 8.1."
                + "<p>Crew data comes from the very awesome Star Trek Timelines Wiki (<a href=https://stt.wiki>https://stt.wiki</a>) and its awesome users.</p>"
                + "<p>App icon by Memory Alpha (<a href=http://memory-alpha.wikia.com>http://memory-alpha.wikia.com</a>).</p>"
                + "<p>Roddenberry font by Jaynz (<a href=http://www.fontspace.com/pixel-sagas/roddenberry>http://www.fontspace.com/pixel-sagas/roddenberry</a>).</p>"//
                + "<p>Icons from the Noun Project (<a href=https://thenounproject.com>https://thenounproject.com</a>): Info by Karthick Nagarajan, Export by Mourad Mokrane, Refresh by Alex Auda Samora, Download by Guillaume Bahri, Zoom out by useiconic.com, Zoom in by useiconic.com, Reset zoom by Leonardo Schneider, Font by iconsmind.com, Help by Dima Lagunov, Log by Arthur Shlain, Stats by Calvin Goodman, Charts by Prerak Patel, About by Gregor Črešnar, Skin by Rflor, Exit by Andres Gleixner, Tables by Viktor Vorobyev, Search by Iconfactory Team, Settings by Michal Beno.</p>"
                + "<p>JFontChooser by Masahiko Sawai (<a href=http://jfontchooser.osdn.jp/site/jfontchooser>http://jfontchooser.osdn.jp/site/jfontchooser</a>).</p>"
                + "<p>Nimbus checkbox alternate row background color bug fix by Harald Barsnes (<a href=https://github.com/compomics>https://github.com/compomics</a>).</p>"
                + "<p>Many thanks to all the awesome people at Stack Overflow (<a href=http://stackoverflow.com>http://stackoverflow.com</a>) and Coderunch (<a href=https://coderanch.com>https://coderanch.com</a>), without all of the excellent information there this project would have been much harder to develop.</p>"
                + "<p>This software is released under the GNU General Public License, version 3 (<a href=https://www.gnu.org/licenses/gpl-3.0.html>https://www.gnu.org/licenses/gpl-3.0.html</a>).</p>"
                + "<p>Game content and materials are trademarks and copyrights of their respective publisher and its licensors.</p>";
        
        aboutEditorPane.setText(text);
        aboutEditorPane.addHyperlinkListener(new HyperlinkListener() {
            public void hyperlinkUpdate(HyperlinkEvent e) {
                if(e.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
                    try {
                        open(e.getURL().toURI());
                    } catch (URISyntaxException ex) {
                        Logger.getLogger(AboutJFrame.class.getName()).log(Level.SEVERE, null, ex);
                    }
                }
            }
        });
    }

    /**
     * This method is called from within the constructor to initialize the form.
     * WARNING: Do NOT modify this code. The content of this method is always
     * regenerated by the Form Editor.
     */
    @SuppressWarnings("unchecked")
    // <editor-fold defaultstate="collapsed" desc="Generated Code">//GEN-BEGIN:initComponents
    private void initComponents() {

        aboutLabel = new javax.swing.JLabel();
        okButton = new javax.swing.JButton();
        beerButton = new javax.swing.JButton();
        logosLabel = new javax.swing.JLabel();
        logoLabel = new javax.swing.JLabel();
        aboutScrollPane = new javax.swing.JScrollPane();
        aboutEditorPane = new javax.swing.JEditorPane();

        setDefaultCloseOperation(javax.swing.WindowConstants.DISPOSE_ON_CLOSE);
        setTitle("About STT Collection Manager");
        setPreferredSize(new java.awt.Dimension(1024, 730));
        setResizable(false);

        aboutLabel.setFont(new java.awt.Font("Tahoma", 1, 11)); // NOI18N
        aboutLabel.setText("About STT Collection Manager v1.2");

        okButton.setText("Ok");
        okButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                okButtonActionPerformed(evt);
            }
        });

        beerButton.setIcon(new javax.swing.ImageIcon(getClass().getResource("/sttcollectionmanager/resources/beer32x32.png"))); // NOI18N
        beerButton.setText("Beer! :)");
        beerButton.setToolTipText("If you like this Software feel free to buy me a beer!");
        beerButton.addActionListener(new java.awt.event.ActionListener() {
            public void actionPerformed(java.awt.event.ActionEvent evt) {
                beerButtonActionPerformed(evt);
            }
        });

        logosLabel.setIcon(new javax.swing.ImageIcon(getClass().getResource("/sttcollectionmanager/resources/logos2.png"))); // NOI18N

        logoLabel.setIcon(new javax.swing.ImageIcon(getClass().getResource("/sttcollectionmanager/resources/appLogoSmallNew.png"))); // NOI18N

        aboutEditorPane.setEditable(false);
        aboutEditorPane.setContentType("text/html"); // NOI18N
        aboutScrollPane.setViewportView(aboutEditorPane);

        javax.swing.GroupLayout layout = new javax.swing.GroupLayout(getContentPane());
        getContentPane().setLayout(layout);
        layout.setHorizontalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(logoLabel)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, 268, Short.MAX_VALUE)
                        .addComponent(logosLabel))
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(aboutLabel)
                        .addGap(0, 0, Short.MAX_VALUE))
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(okButton, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                        .addComponent(beerButton)))
                .addContainerGap())
            .addComponent(aboutScrollPane, javax.swing.GroupLayout.Alignment.TRAILING)
        );
        layout.setVerticalGroup(
            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup()
                .addContainerGap()
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup()
                        .addComponent(logoLabel)
                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                        .addComponent(aboutLabel))
                    .addComponent(logosLabel))
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addComponent(aboutScrollPane, javax.swing.GroupLayout.DEFAULT_SIZE, 385, Short.MAX_VALUE)
                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
                    .addComponent(okButton, javax.swing.GroupLayout.Alignment.TRAILING)
                    .addComponent(beerButton, javax.swing.GroupLayout.Alignment.TRAILING))
                .addContainerGap())
        );

        pack();
    }// </editor-fold>//GEN-END:initComponents

    private void okButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_okButtonActionPerformed
        this.dispose();
    }//GEN-LAST:event_okButtonActionPerformed

    private void beerButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_beerButtonActionPerformed
        URI uri = null;
        try {
            uri = new URI("https://www.paypal.me/augustenz");
        } catch (URISyntaxException ex) {
            Logger.getLogger(AboutJFrame.class.getName()).log(Level.SEVERE, null, ex);
        }
        open(uri);
    }//GEN-LAST:event_beerButtonActionPerformed

    /**
     * @param args the command line arguments
     */
    public static void main(String args[]) {
        /* Set the Nimbus look and feel */
        //<editor-fold defaultstate="collapsed" desc=" Look and feel setting code (optional) ">
        /* If Nimbus (introduced in Java SE 6) is not available, stay with the default look and feel.
         * For details see http://download.oracle.com/javase/tutorial/uiswing/lookandfeel/plaf.html 
         */
        try {
            for (javax.swing.UIManager.LookAndFeelInfo info : javax.swing.UIManager.getInstalledLookAndFeels()) {
                if ("Nimbus".equals(info.getName())) {
                    javax.swing.UIManager.setLookAndFeel(info.getClassName());
                    break;
                }
            }
        } catch (ClassNotFoundException ex) {
            java.util.logging.Logger.getLogger(AboutJFrame.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (InstantiationException ex) {
            java.util.logging.Logger.getLogger(AboutJFrame.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (IllegalAccessException ex) {
            java.util.logging.Logger.getLogger(AboutJFrame.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        } catch (javax.swing.UnsupportedLookAndFeelException ex) {
            java.util.logging.Logger.getLogger(AboutJFrame.class.getName()).log(java.util.logging.Level.SEVERE, null, ex);
        }
        //</editor-fold>
        //</editor-fold>

        /* Create and display the form */
        java.awt.EventQueue.invokeLater(new Runnable() {
            public void run() {
                new AboutJFrame().setVisible(true);
            }
        });
    }

    // Variables declaration - do not modify//GEN-BEGIN:variables
    private javax.swing.JEditorPane aboutEditorPane;
    private javax.swing.JLabel aboutLabel;
    private javax.swing.JScrollPane aboutScrollPane;
    private javax.swing.JButton beerButton;
    private javax.swing.JLabel logoLabel;
    private javax.swing.JLabel logosLabel;
    private javax.swing.JButton okButton;
    // End of variables declaration//GEN-END:variables
    private void open(URI uri) {
        if (Desktop.isDesktopSupported()) {
          Desktop desktop = Desktop.getDesktop();
          try {
            desktop.browse(uri);
          } catch (IOException e) {
            JOptionPane.showMessageDialog(this,
                "Failed to launch the link, your computer is likely misconfigured.",
                "Cannot Launch Link",JOptionPane.WARNING_MESSAGE);
          }
        } else {
          JOptionPane.showMessageDialog(this,
              "Java is not able to launch links on your computer.",
              "Cannot Launch Link", JOptionPane.WARNING_MESSAGE);
        }
    }  
}

